From 37aaef02ee9cb60a71da1b92b6ff5cd515537fc0 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:31 -0400
Subject: [usb] wusbcore: adjust iterator correctly when searching for ep comp descriptor

Message-id: <1410887463-20674-140-git-send-email-dzickus@redhat.com>
Patchwork-id: 95411
O-Subject: [RHEL7 PATCH 139/591] usb: wusbcore: adjust iterator correctly when searching for ep comp descriptor
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 4659a2452baa7d89324fda097158d7f8fe71e0cb
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Fri Feb 28 15:15:19 2014 -0600

    usb: wusbcore: adjust iterator correctly when searching for ep comp descriptor

    If the endpoint companion descriptor is not the first descriptor in the
    extra descriptor buffer of a usb_host_endpoint, the loop in
    rpipe_epc_find will get its buffer pointer and remaining size values out
    of sync.  The buffer ptr 'itr' is advanced by the descriptor's bLength
    field but the remaining size value 'itr_size' is decremented by the
    bDescriptorType field which is incorrect.  This patch fixes the loop to
    decrement itr_size by bLength as it should.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-rpipe.c b/drivers/usb/wusbcore/wa-rpipe.c
index 4f80f7c..91422c8 100644
--- a/drivers/usb/wusbcore/wa-rpipe.c
+++ b/drivers/usb/wusbcore/wa-rpipe.c
@@ -299,7 +299,7 @@ static struct usb_wireless_ep_comp_descriptor *rpipe_epc_find(
    break;
   }
   itr += hdr->bLength;
-  itr_size -= hdr->bDescriptorType;
+  itr_size -= hdr->bLength;
  }
 out:
  return epcd;
-- 
1.7.1